backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -1 +1,44 @@
-export default class TeamEventAttendanceDao {}
+import { v4 as uuidv4 } from 'uuid';
+import { memberCollection, teamEventAttendanceCollection } from '../firebase';
+import { DBTeamEventAttendance } from '../types/DataTypes';
+
+export default class TeamEventAttendanceDao {
+  static async createTeamEventAttendance(
+    teamEventAttendance: TeamEventAttendance
+  ): Promise<TeamEventAttendance> {
+    const teamAttendanceRef: DBTeamEventAttendance = {
+      member: memberCollection.doc(teamEventAttendance.member.email),
+      ...(typeof teamEventAttendance.hoursAttended === 'number'
+        ? { hoursAttended: teamEventAttendance.hoursAttended }
+        : {}),
+      image: teamEventAttendance.image,
+      eventUuid: teamEventAttendance.eventUuid,
+      pending: true,
+      uuid: teamEventAttendance.uuid ? teamEventAttendance.uuid : uuidv4()
```

Comment:
+1 to Henry's comment, applied here too. You can just use the spread operator here, and then any fields that don't apply you can just assign them afterwards.

Look at `TeamEventsDao.createTeamEvent` for an example of that syntax. Fields that you specify explictly after the spread are "overwritten".

```
      requests: event.requests.map((req) => ({
        ...req,
        member: memberCollection.doc(req.member.email)
      })),
      attendees: event.attendees.map((att) => ({
        ...att,
        member: memberCollection.doc(att.member.email)
      }))
```